MapboxNavigationTests/InstructionsBannerViewIntegrationTests.swift

```diff
@@ -89,33 +90,83 @@ class InstructionsBannerViewIntegrationTests: XCTestCase {
+        
+        //ensure that first callback fires
+        wait(for: [firstExpectation], timeout: 5)
 
+        //change the callback to track the second shield component
+        view.primaryLabel.imageDownloadCompletion = secondExpectation.fulfill
+        
         let secondDestinationComponent = instructions[2]
         simulateDownloadingShieldForComponent(secondDestinationComponent)
 
+        //ensure that second callback fires
+        wait(for: [secondExpectation], timeout: 5)
+ 
         //Slash should no longer be present
         XCTAssertNil(view.primaryLabel.text!.index(of: "/"), "Expected instruction text not to contain a slash: \(view.primaryLabel.text!)")
     }
+    
+    func testExitBannerIntegration() {
+        let exitAttribute = VisualInstructionComponent(type: .exit, text: "Exit", imageURL: nil,  abbreviation: nil, abbreviationPriority: 0)
+        let exitCodeAttribute = VisualInstructionComponent(type: .exitCode, text: "123A", imageURL: nil, abbreviation: nil, abbreviationPriority: 0)
+        let mainStreetString = VisualInstructionComponent(type: .text, text: "Main Street", imageURL: nil, abbreviation: "Main St", abbreviationPriority: 0)
+        let exitInstruction = VisualInstruction(text: nil, maneuverType: .takeOffRamp, maneuverDirection: .right, textComponents: [exitAttribute, exitCodeAttribute, mainStreetString])
+        
+        let label = InstructionLabel(frame: CGRect(origin: .zero, size:CGSize(width: 375, height: 100)))
+        
+        label.availableBounds = { return label.frame }
+        
+        let presenter = InstructionPresenter(exitInstruction, dataSource: label, downloadCompletion: nil)
+        let attributed = presenter.attributedText()
+        
+        let spaceRange = NSMakeRange(1, 1)
+        let space = attributed.attributedSubstring(from: spaceRange)
+        //Do we have spacing between the attachment and the road name?
+        XCTAssert(space.string == " ", "Should be a space between exit attachment and name")
+        
+        //Road Name should be present and not abbreviated
+        XCTAssert(attributed.length == 13, "Road name should not be abbreviated")
+        
+        let roadNameRange = NSMakeRange(2, 11)
+        let roadName = attributed.attributedSubstring(from: roadNameRange)
+        XCTAssert(roadName.string == "Main Street", "Banner not populating road name correctly")
+    }
 
     private func simulateDownloadingShieldForComponent(_ component: VisualInstructionComponent) {
         let operation: ImageDownloadOperationSpy = ImageDownloadOperationSpy.operationForURL(component.imageURL!)!
         operation.fireAllCompletions(ShieldImage.i280.image, data: UIImagePNGRepresentation(ShieldImage.i280.image), error: nil)
+        //FIXME: need to get to a place where this isn't necessary, probably by adjusting our test code.
+        RunLoop.current.run(until: Date(timeIntervalSinceNow: 1))
```

Comment:
This is OK for the moment but I would like to eliminate the need for this shortly after we merge. This is mostly a note to self.